### PR TITLE
Feature/skeleton component

### PR DESCRIPTION
**Pull request by susox**
### Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### Description

Yup `precise-ui` provides ProgressBar and Spinner components. Perfect for their specific cases. However, there are times where we load dynamic content and the page is simply mmmmm well blank. Kind of ugly actually 🤮... Not anymore! 

Now there is an awesome 🦄new component in town!. It is flexible enough to adapt to many custom layouts. I can also see current components being refactored to use `Skeletons` while the actual data arrives, e.g.: .

Check out the following examples 😲😎🤙🚀:

skeleton-2

### CODE REVIEW

File: docs/introduction/components.md [hunk 1]
```diff
@@ -15,10 +15,10 @@
 - [x] Button
 - [x] Card
 - [x] Checkbox
-- [x] Content Switcher
```
Comment (guiherzog) [inline docs/introduction/components.md]: OCD!

File: src/components/Skeleton/Example.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
...v84dr97en7wI74o4/6vK781s384tz+45T4nYf5saH93p76uYz6t4z+45X6vY38ypD+5ZX4oYj5sYv7x4/4p4n+5pX+6pb5rIr935P93pT4n4f805L7yI/7xo/5qIn6uIz915L4noj7xI791ZL4n4j8z5H92ZP4ooj925P+6ZX6vI34pIj80JH80pH4pIn6vo76tYz93ZT93JP+4ZT7v477wo78zpD81JL+6Jb7vo35r4r4pon6toz+4JT8zJD7xY/6uo36uY37v4392pPZ0b/yAAACmklEQVR4AezWg5UEQRRA0bZtr51/fOuxDqca7yZQ+JQAAAAAAAAAAAAAAAAwI7KiapqqyNJI6Ib5y9BHch9zbRQ3kg1zzRhD1BRziyKJZ5lbVEk8zdyicaFDqrlFHX9SU/anGyOjY0NWrM1wxTixZ9FO2LPYs2zHHdeFPN8PwihO0mwke1bu/yvKqm7aTnhS1/6Ofri5vXPuBZb9g39M8JjHT6mQxvjsn1aUL69v7ft1R8eHf0k/fPFaT+kVRDEAgPPeL7Vt27at63Ft27bNrdZ2mzP9dxAn1DMsPOLf/qxI/JmoaLsYv1hQXxz+RnxCYlJQcjCoyBJ/4bmKKalpEaCKMPyz9OiMmMwQYCwLiSLjspNyknOBGXNkIS+/oLCITRWLS0rLyisqkYn0qoxq/xogqdXc0er0BqOJ45GByAwgEDSviJKs1NU3NCJFDlA0aT7S3NLa1t7R2YV/kA8k3Zqv9PT29Q8MDg3jL4wAyajmB8bGJyanpmdm3X5yZVyBZk7zc/MLi0vLK6tr+AUnIKrV/N76xubWdvnOR9vC3gyIdvc0f6bV7R8cHh3z9AS9dKIhO5XOlPOL222RHwxkl9ft1LVBBFEABNBpclJiargydiOSc3d397vFHaohxW3no/8V8UJU2Q1HIBClTAwKgRBF3DgkEhRJQiOVpkQGKlkq5PJQKRTpn1uCTrlC36pQqtGvOrQygoGk8g360SxALZDm5yVgQKvNz+rAiG5PGrRAy+UnhPowZjDkh+VGMGinyA9qOjBqPOFHTGcwbr7guy0dfAFnL8R3Wa3xRcrNEN+02uALbT2Xr1nsb/HFnINVj8+bTg4L+A5H8+MTlw+dntVG+Fbn84tL72oVu57c1Ga3KfxUlmVZlmXdAber1Qd03DKcAAAAAElFTkSuQmCC';
```
Comment (guiherzog) [inline src/components/Skeleton/Example.md]: Moving this image to another file or to a public cdn would make the example a bit nicer.

File: src/components/Skeleton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+      ${height &&
+        width &&
+        isCircle && {
+          borderRadius: '50%',
+        }};
+    `;
+
+    const Span = styled.span`
+      background-color: ${defaultBaseColor};
```
Comment (guiherzog) [inline src/components/Skeleton/index.tsx]: Maybe moving this style outside of the for loop or the component (and composing it with the extra style) would be better.

File: src/components/Skeleton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+export const Skeleton: React.FC<SkeletonProps> = props => {
+  const { count = 1, duration = 1.2, width = '100%', height = '100%', isCircle, isText, isPulsing = true } = props;
+  const skeletons = [];
+
+  for (let i = 0; i < count; i++) {
+    const extraStyle = css`
+      animation: ${shine} ${duration}s infinite linear ${!isPulsing ? 'paused' : 'running'};
+
+      ${count > 1 && isText
+        ? { width: `${Math.floor(Math.random() * (100 - 80 + 1) + 80)}%` }
```
Comment (boromo) [inline src/components/Skeleton/index.tsx]: What is this magical formula doing - `(100 - 80 + 1)`? 🍔

File: src/components/Skeleton/index.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        width &&
+        isCircle && {
+          borderRadius: '50%',
+        }};
+    `;
+
+    const Span = styled.span`
+      background-color: ${defaultBaseColor};
+      background-image: linear-gradient(90deg, ${defaultBaseColor}, ${defaultHighlightColor}, ${defaultBaseColor});
+      background-size: 60px 100%;
```
Comment (boromo) [inline src/components/Skeleton/index.tsx]: How did you get this 60px value, somehow doesn't seem right. Shouldn't it be somehow relative to something?

File: src/components/Skeleton/index.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      ${height &&
+        width &&
+        isCircle && {
+          borderRadius: '50%',
+        }};
+    `;
+
+    const Span = styled.span`
+      background-color: ${defaultBaseColor};
+      background-image: linear-gradient(90deg, ${defaultBaseColor}, ${defaultHighlightColor}, ${defaultBaseColor});
```
Comment (boromo) [inline src/components/Skeleton/index.tsx]: What about theming the colors?

File: src/components/Skeleton/Example.md (hunk 2 above)
Comment (susox) [reply src/components/Skeleton/Example.md]: Just using the same as the Avatar component example ;) (https://precise-ui.io/#/Components/Avatar). But I see your point, I'll get another one online.

File: src/components/Skeleton/index.tsx (hunk 4 above)
Comment (IKolosynskyi) [reply src/components/Skeleton/index.tsx]: Looks like width could be above 100% - 0.99*21+80 ~ 100.79

File: src/components/Skeleton/index.tsx (hunk 5 above)
Comment (susox) [reply src/components/Skeleton/index.tsx]: well that is the width of the shine/pulse, nothing magic about it. Would you prefer this value to be exposed as a prop?, e.g.: 'pulseWidth={60}'? 🧐

File: src/components/Skeleton/index.tsx (hunk 4 above)
Comment (susox) [reply src/components/Skeleton/index.tsx]: just a random number between 80 and 100 to simulated different text lines

File: src/components/Skeleton/Example.md (hunk 2 above)
Comment (susox) [reply src/components/Skeleton/Example.md]: done

File: src/components/Skeleton/index.tsx (hunk 3 above)
Comment (susox) [reply src/components/Skeleton/index.tsx]: composed!

File: src/components/Skeleton/index.tsx (hunk 4 above)
Comment (susox) [reply src/components/Skeleton/index.tsx]: rewritten ;)

File: src/components/Skeleton/index.tsx (hunk 6 above)
Comment (susox) [reply src/components/Skeleton/index.tsx]: themed!

File: src/components/Skeleton/index.tsx [hunk 7]
```diff
@@ -39,28 +38,37 @@ export interface SkeletonProps extends StandardProps {
...
   const skeletons = [];
 
   for (let i = 0; i < count; i++) {
-    const extraStyle = css`
+    const StyledSpan = styled(Span)`
       animation: ${shine} ${duration}s infinite linear ${!isPulsing ? 'paused' : 'running'};
 
       ${count > 1 && isText
-        ? { width: `${Math.floor(Math.random() * (100 - 80 + 1) + 80)}%` }
+        ? { width: `${Math.floor(Math.random() * 21 + 80)}%` } // random number between 80 and 100
```
Comment (dimabory) [inline src/components/Skeleton/index.tsx]: I would create the function and use it here like `randInt(80, 100)`, which is more clear than just a comment. 

```ts
const randInt = (min: number, max: number) => 
	Math.floor(Math.random() * (max - min + 1)) + min;

**Pull request merged**